extraction/src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineBuilder.scala

```diff
@@ -24,52 +21,86 @@ import scala.concurrent.Future
 class ExtractionPipelineBuilder(getClient: () => EncodeClient)
     extends PipelineBuilder[Args]
     with Serializable {
+  import ExtractionPipelineBuilder._
+
   implicit val coder: Coder[Msg] = Coder.beam(new UpackMsgCoder)
 
+  /** Convenience alias to make signatures a little less nasty. */
+  private type LookupParams = (EncodeEntity, List[(String, String)])
+
+  private val lookupFn =
```

Comment:
This used to be a `def` that would produce one lookup function per encode entity. Refactored to push the entity types into the `asyncLookup` args so we could enforce a global concurrency limit.